File: fabric-backend-core/src/main/java/io/storydoc/fabric/query/app/ResultDTO.java
package io.storydoc.fabric.query.app;

import io.storydoc.fabric.console.app.navigation.NavItem;
import io.storydoc.fabric.query.app.documents.DocumentsResultSet;
import io.storydoc.fabric.query.app.tabular.TabularResultSet;
import io.storydoc.fabric.query.app.tabular.TabularResultSetMetaDataDTO;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ResultDTO {
    private String systemType;
    private ResultType resultType;
    private TabularResultSetMetaDataDTO description;
    private String content;
    private DocumentsResultSet documentsResultSet;
    private TabularResultSet tabular;
    private List<NavItem> navItems;
}

File: fabric-frontend/src/app/api/models/result-dto.ts
/* tslint:disable */
/* eslint-disable */
import { DocumentsResultSet } from './documents-result-set';
import { NavItem } from './nav-item';
import { TabularResultSet } from './tabular-result-set';
import { TabularResultSetMetaDataDto } from './tabular-result-set-meta-data-dto';
export interface ResultDto {
  content?: string;
  description?: TabularResultSetMetaDataDto;
  documentsResultSet?: DocumentsResultSet;
  navItems?: Array<NavItem>;
  resultType?: 'JSON' | 'MESSAGE' | 'STACKTRACE' | 'TABULAR';
  systemType?: string;
  tabular?: TabularResultSet;
}
